File: clients/algoliasearch-client-php/lib/Model/Search/HasPendingMappingsResponse.php
<?php

namespace Algolia\AlgoliaSearch\Model\Search;

/**
 * HasPendingMappingsResponse Class Doc Comment
 *
 * @category Class
 * @package Algolia\AlgoliaSearch
 */
class HasPendingMappingsResponse extends \Algolia\AlgoliaSearch\Model\AbstractModel implements
        ModelInterface,
        \ArrayAccess,
        \JsonSerializable
{
    /**
     * Array of property to type mappings. Used for (de)serialization
     *
     * @var string[]
     */
    protected static $modelTypes = [
        'pending' => 'bool',
        'clusters' => 'array<string,string[]>',
    ];

    /**
     * Array of property to format mappings. Used for (de)serialization
     *
     * @var string[]
     */
    protected static $modelFormats = [
        'pending' => null,
        'clusters' => null,
    ];

    /**
     * Array of property to type mappings. Used for (de)serialization
     *
     * @return array
     */
    public static function modelTypes()
    {
        return self::$modelTypes;
    }

    /**
     * Array of property to format mappings. Used for (de)serialization
     *
     * @return array
     */
    public static function modelFormats()
    {
        return self::$modelFormats;
    }

    /**
     * Array of attributes to setter functions (for deserialization of responses)
     *
     * @var string[]
     */
    protected static $setters = [
        'pending' => 'setPending',
        'clusters' => 'setClusters',
    ];

    /**
     * Array of attributes to getter functions (for serialization of requests)
     *
     * @var string[]
     */
    protected static $getters = [
        'pending' => 'getPending',
        'clusters' => 'getClusters',
    ];

    /**
     * Array of attributes to setter functions (for deserialization of responses)
     *
     * @return array
     */
    public static function setters()
    {
        return self::$setters;
    }

    /**
     * Array of attributes to getter functions (for serialization of requests)
     *
     * @return array
     */
    public static function getters()
    {
        return self::$getters;
    }

    /**
     * Associative array for storing property values
     *
     * @var mixed[]
     */
    protected $container = [];

    /**
     * Constructor
     *
     * @param mixed[] $data Associated array of property values
     */
    public function __construct(array $data = null)
    {
        if (isset($data['pending'])) {
            $this->container['pending'] = $data['pending'];
        }
        if (isset($data['clusters'])) {
            $this->container['clusters'] = $data['clusters'];
        }
    }

    /**
     * Show all the invalid properties with reasons.
     *
     * @return array invalid properties with reasons
     */
    public function listInvalidProperties()
    {
        $invalidProperties = [];

        if (
            !isset($this->container['pending']) ||
            $this->container['pending'] === null
        ) {
            $invalidProperties[] = "'pending' can't be null";
        }

        return $invalidProperties;
    }

    /**
     * Validate all the properties in the model
     * return true if all passed
     *
     * @return bool True if all properties are valid
     */
    public function valid()
    {
        return count($this->listInvalidProperties()) === 0;
    }

    /**
     * Gets pending
     *
     * @return bool
     */
    public function getPending()
    {
        return $this->container['pending'] ?? null;
    }

    /**
     * Sets pending
     *
     * @param bool $pending if there is any clusters with pending mapping state
     *
     * @return self
     */
    public function setPending($pending)
    {
        $this->container['pending'] = $pending;

        return $this;
    }

    /**
     * Gets clusters
     *
     * @return array<string,string[]>|null
     */
    public function getClusters()
    {
        return $this->container['clusters'] ?? null;
    }

    /**
     * Sets clusters
     *
     * @param array<string,string[]>|null $clusters describe cluster pending (migrating, creating, deleting) mapping state
     *
     * @return self
     */
    public function setClusters($clusters)
    {
        $this->container['clusters'] = $clusters;

        return $this;
    }
    /**
     * Returns true if offset exists. False otherwise.
     *
     * @param int $offset Offset
     *
     * @return bool
     */
    public function offsetExists($offset)
    {
        return isset($this->container[$offset]);
    }

    /**
     * Gets offset.
     *
     * @param int $offset Offset
     *
     * @return mixed|null
     */
    public function offsetGet($offset)
    {
        return $this->container[$offset] ?? null;
    }

    /**
     * Sets value based on offset.
     *
     * @param int|null $offset Offset
     * @param mixed    $value  Value to be set
     *
     * @return void
     */
    public function offsetSet($offset, $value)
    {
        if (is_null($offset)) {
            $this->container[] = $value;
        } else {
            $this->container[$offset] = $value;
        }
    }

    /**
     * Unsets offset.
     *
     * @param int $offset Offset
     *
     * @return void
     */
    public function offsetUnset($offset)
    {
        unset($this->container[$offset]);
    }
}

File: clients/algoliasearch-client-java-2/algoliasearch-core/src/main/java/com/algolia/model/search/HasPendingMappingsResponse.java
package com.algolia.model.search;

import com.fasterxml.jackson.annotation.*;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Objects;

/** HasPendingMappingsResponse */
public class HasPendingMappingsResponse {

  @JsonProperty("pending")
  private Boolean pending;

  @JsonProperty("clusters")
  private Map<String, List<String>> clusters;

  public HasPendingMappingsResponse setPending(Boolean pending) {
    this.pending = pending;
    return this;
  }

  /**
   * If there is any clusters with pending mapping state.
   *
   * @return pending
   */
  @javax.annotation.Nonnull
  public Boolean getPending() {
    return pending;
  }

  public HasPendingMappingsResponse setClusters(Map<String, List<String>> clusters) {
    this.clusters = clusters;
    return this;
  }

  public HasPendingMappingsResponse putClusters(String key, List<String> clustersItem) {
    if (this.clusters == null) {
      this.clusters = new HashMap<>();
    }
    this.clusters.put(key, clustersItem);
    return this;
  }

  /**
   * Describe cluster pending (migrating, creating, deleting) mapping state.
   *
   * @return clusters
   */
  @javax.annotation.Nullable
  public Map<String, List<String>> getClusters() {
    return clusters;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    HasPendingMappingsResponse hasPendingMappingsResponse = (HasPendingMappingsResponse) o;
    return (
      Objects.equals(this.pending, hasPendingMappingsResponse.pending) && Objects.equals(this.clusters, hasPendingMappingsResponse.clusters)
    );
  }

  @Override
  public int hashCode() {
    return Objects.hash(pending, clusters);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class HasPendingMappingsResponse {\n");
    sb.append("    pending: ").append(toIndentedString(pending)).append("\n");
    sb.append("    clusters: ").append(toIndentedString(clusters)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}
